### PR TITLE
new(userspace/libsinsp): support plugins in sinsp-example.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area libsinsp

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

Expands sinsp-example to support plugins.
This will be useful to test plugins; moreover in the future when container metadata support will become a plugin(https://github.com/falcosecurity/falco/issues/3403), this will be needed to run tests in CI (like for example e2e tests).

To pass init config/open params, i chose to use a single string following this pattern:
`filepath.so|{initconfig}|{openParams}`, eg:
```
./libsinsp/examples/sinsp-example -p "/home/federico/Scaricati/dummy-0.11.4-linux-x86_64/libdummy.so|{\"jitter\": 10}" -f dummy.divisible[4]=1
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libsinsp/examples/test.cpp [hunk 1]
```diff
@@ -55,13 +55,16 @@ static bool ppm_sc_modifies_state = false;
 static bool ppm_sc_repair_state = false;
 static bool ppm_sc_state_remove_io_sc = false;
 static bool enable_glogger = false;
-static string engine_string = KMOD_ENGINE; /* Default for backward compatibility. */
+static string engine_string;
```
Comment (FedeDP) [inline userspace/libsinsp/examples/test.cpp:58]: This is initialized below when empty.
This way, we can check if another engine was already set and give an error.

File: userspace/libsinsp/examples/test.cpp [hunk 2]
```diff
@@ -467,6 +536,8 @@ void formatted_dump(sinsp&, sinsp_evt* ev) {
 	} else if(ev->get_category() == EC_NET || ev->get_category() == EC_IO_READ ||
 	          ev->get_category() == EC_IO_WRITE) {
 		net_formatter->tostring(ev, output);
+	} else if(ev->get_info()->category & EC_PLUGIN) {
```
Comment (FedeDP) [inline userspace/libsinsp/examples/test.cpp:539]: We can't use `ev->get_category()` because that only works for syscall-based events.

**Comment by FedeDP** [comment]
need to fix e2e tests.

**Review by jasondellaluce — APPROVED** [review]

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**